tests/id/id.cpp

```diff
@@ -42,11 +42,6 @@ int main() {
   // Test that id<> implement comparable
   if (jj != make_id({ 5, 7 }))
     std::cout << "jj has changed" << std::endl;
-  if (jj > make_id({ 5, 7 }))
-    std::cout << "jj is lexicographically greater than before" << std::endl;
-  if (jj < make_id(5, 7))
-    std::cout << "jj is lexicographically lesser than before" << std::endl;
```

Comment:
After reading more carefully the spec, the comparison operations are element-wise instead of being lexicographic as provided by `std::array`. So it might give more sense to some of your changes. :-)
Probably we should have some test to display the vector of boolean in result instead.

Reply:
Yes there is no bool returned comparison which is why I removed anything that handled that. Doing a bool conversion is quite messy so I'm not too surprised the spec didn't add it. I actually had code to do it locally using BOOST_EXPLICIT_OPERATOR_BOOL() however the whole safe bool idiom scared me, and I think going outside the spec here will introduce more ambiguity problems.

Reply:
the spec also says the results id should be bools casted to size_t.